zipkin-server/src/test/java/zipkin2/server/internal/elasticsearch/ITElasticsearchDynamicCredentials.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.elasticsearch;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit.server.mock.MockWebServerExtension;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import zipkin2.elasticsearch.ElasticsearchStorage;
+
+import static java.time.Duration.ofSeconds;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTimeout;
+import static zipkin2.elasticsearch.Access.pretendIndexTemplatesExist;
+import static zipkin2.server.internal.elasticsearch.TestResponses.YELLOW_RESPONSE;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.PASSWORD_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.USERNAME_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageProperties.Ssl;
+
+class ITElasticsearchDynamicCredentials {
+
+  @RegisterExtension static MockWebServerExtension server = new MockWebServerExtension() {
+    @Override protected void configureServer(ServerBuilder sb) throws Exception {
+      sb.https(0);
+      Ssl ssl = new Ssl();
+      ssl.setKeyStore("classpath:keystore.jks");
+      ssl.setKeyStorePassword("password");
+      ssl.setTrustStore("classpath:keystore.jks");
+      ssl.setTrustStorePassword("password");
+
+      final KeyManagerFactory keyManagerFactory = SslUtil.getKeyManagerFactory(ssl);
+      final TrustManagerFactory trustManagerFactory = SslUtil.getTrustManagerFactory(ssl);
+      sb.tls(keyManagerFactory)
+        .tlsCustomizer(sslContextBuilder -> {
+          sslContextBuilder.keyManager(keyManagerFactory);
+          sslContextBuilder.trustManager(trustManagerFactory);
+        });
+    }
+  };
+
+  AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+  ElasticsearchStorage storage;
+  File credentialsFile;
+
+  @BeforeEach void init() throws IOException {
+    credentialsFile = File.createTempFile("zipkin-server", "credentials");
```

Comment:
I have added two files, `es-credentials` and `es-credentials`. They are loaded in `ZipkinElasticsearchStorageConfigurationTest` and `ITElasticsearchDynamicCredentials` respectively. We can't share a single file between them because the later one will change the content of the file, which may cause the first test failed.